File: src/main/java/me/nasukhov/TukitaLearner/config/AppConfig.java
package me.nasukhov.TukitaLearner.config;

import me.nasukhov.TukitaLearner.Updater;
import me.nasukhov.TukitaLearner.bot.bridge.tg.Telegram;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;

@Configuration
@ComponentScan(basePackages = "me.nasukhov.TukitaLearner")
public class AppConfig {
    @Bean
    public CommandLineRunner createLauncher(Updater updater, Telegram telegram) {
        return (args) -> {
            try {
                updater.execute();

                telegram.run();
            } catch (Throwable e) {
                // TODO
                e.printStackTrace();
            }
        };
    }
}

File: src/main/kotlin/me/nasukhov/tukitalearner/config/AppConfig.kt
package me.nasukhov.tukitalearner.config

import me.nasukhov.bot.adapter.tg.Telegram
import me.nasukhov.tukitalearner.Updater
import org.springframework.boot.CommandLineRunner
import org.springframework.cache.annotation.EnableCaching
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.core.annotation.Order

@Configuration
@EnableCaching
@ComponentScan(
    basePackages = [
        "me.nasukhov.bot",
        "me.nasukhov.utility",
        "me.nasukhov.tukitalearner",
    ],
)
class AppConfig {
    @Bean
    @Profile("!test")
    @Order(1)
    fun createUpdater(updater: Updater): CommandLineRunner = CommandLineRunner { updater.execute() }

    @Bean
    @Order(2)
    @Profile("!test")
    fun createLauncher(telegram: Telegram): CommandLineRunner = CommandLineRunner { telegram.run() }
}
